src/Nix/Render.hs

```diff
@@ -10,26 +10,22 @@
 module Nix.Render where
 
 import           Data.ByteString (ByteString)
-import           Data.List.NonEmpty (NonEmpty((:|)))
 import qualified Data.Set as Set
 import           Data.Void
 import           Nix.Expr.Types.Annotated
 import           Text.Megaparsec.Error
 import           Text.Megaparsec.Pos (SourcePos(..))
 import           Text.PrettyPrint.ANSI.Leijen
 
+
 class Monad m => MonadFile m where
     readFile :: FilePath -> m ByteString
 
-posAndMsg :: SourcePos -> Doc -> ParseError t Void
-posAndMsg beg msg =
-    FancyError (beg :| [])
+posAndMsg :: SourcePos -> Doc -> ParseError s Void
+posAndMsg (SourcePos _ lineNo _) msg =
+    FancyError (unPos lineNo)
         (Set.fromList [ErrorFail (show msg) :: ErrorFancy Void])
 
-renderLocation :: MonadFile m => SrcSpan -> Doc -> m Doc
-renderLocation (SrcSpan beg@(SourcePos "<string>" _ _) _) msg =
-    return $ text $ init $ parseErrorPretty @Char (posAndMsg beg msg)
-
-renderLocation (SrcSpan beg@(SourcePos path _ _) _) msg = do
-    contents <- Nix.Render.readFile path
-    return $ text $ init $ parseErrorPretty' contents (posAndMsg beg msg)
+renderLocation :: Monad m => SrcSpan -> Doc -> m Doc
+renderLocation (SrcSpan beg@(SourcePos _ _ _) _) msg =
```

Comment:
I think we still need to handle `<string>` case

Reply:
How should it be handled differently?

Reply:
Ah, seems like path doesn't need resolving anymore and that was the special case. All good :)